src/main/kotlin/com/cognifide/gradle/aem/pkg/ExtractPackageTask.kt

```diff
@@ -0,0 +1,73 @@
+package com.cognifide.gradle.aem.pkg
+
+import com.cognifide.gradle.aem.api.AemDefaultTask
+import com.cognifide.gradle.aem.base.vlt.VltCleaner
+import com.cognifide.gradle.aem.internal.file.FileException
+import org.gradle.api.tasks.Input
+import org.gradle.api.tasks.Internal
+import org.gradle.api.tasks.TaskAction
+import java.io.File
+
+open class ExtractPackageTask : AemDefaultTask() {
+
+    companion object {
+        const val NAME = "aemDownloadExtract"
+        const val FORCE_NEW_FLAG = "extract.force.new"
+        const val SKIP_CLEANUP = "extract.skip.cleanup"
+    }
+
+    @Input
+    val packageForceNew = props.flag(FORCE_NEW_FLAG)
+
+    @Input
+    val skipCleanup = props.flag(SKIP_CLEANUP)
+
+    @Internal
+    val vltCleaner = VltCleaner(project)
+
+    init {
+        description = "Extracts the contents of package downloaded using ${DownloadTask.NAME} task"
+    }
+
+    @TaskAction
+    fun extract() {
+        val ext = project.extensions.extraProperties
+        if (ext.has(DownloadTask.DOWNLOADED_PACKAGE_PROPERTY)) {
```

Comment:
if extract will be run separatelly, it will not work, exception will be thrown immediatelly. this could not work in this way definitelly.

Reply:
file location should be determined by convention regardless executed separatelly or not

Reply:
Now it's running in single task